src/type-resolution/resolve.ts

```diff
@@ -162,12 +162,12 @@ export function analyzeTypeReference(
   if (isEnumLikeClass(typeRef.type)) {
     return ResolvableExpressionType.ENUM_LIKE_CLASS;
   }
-  if (isSerializableInterface(typeRef.type)) {
-    return ResolvableExpressionType.STRUCT;
-  }
   if (isBehavioralInterface(typeRef.type)) {
     return ResolvableExpressionType.BEHAVIORAL_INTERFACE;
   }
+  if (isSerializableInterface(typeRef.type)) {
```

Comment:
This is bug fix. Many behavioral interfaces are being classified as structs because of the order of evaluation.

Reply:
Do we have an explicit test case for this? If not maybe a comment would be useful.